guide/14-deep-learning/how_pspnet_works_and_its_implementation_details_within_arcgis.ipynb

```diff
@@ -0,0 +1,287 @@
+{
```

Comment:
<blockquote>The feature map from the backbone is pooled at different sizes and then passed through a convolution layer and after which upsampling takes place to make them the same size as of the original feature map. Finally, the upsampled maps are concatenated with the original feature map to be passed to the decoder. This technique fuses the features at four different scales, which you can pass as a parameter to the&nbsp;<code>PSPNetClassifier</code>&nbsp;model. The four levels are a defined from coarse to fine. Suppose if we pool the features at scales of (1, 3, 5, 6). The first is the most coarse feature map as it captures all the information in just 1x1 spatial location, whereas as the spatial resolution increases, the high resolution features are also taken in account as in the case of 6x6 pyramid size. A rule of thumb is that the if the pyramid size is small i.e ~1 or 2, the model will capture the low resolution and bigger features whereas if the the pyramid size is bigger i.e ~6-8, the model will be able to capture high resolution features.&nbsp;So, if we have a large variance in the size of features to be segmented, we might want to increase the number of pyramids and the size of pyramids in the PSPNet module, on the other hand if we have similiar sized features, we can get away with only one or two level pyramid.</blockquote><blockquote><br></blockquote><p> </p><p>The feature map from the backbone is pooled at different sizes and then passed through a convolution layer. After this, up-sampling takes place <strong>on ???</strong> to resize it to original feature map. Finally, the up-sampled feature maps are concatenated with the original feature map to be passed to the decoder. <strong>(can we explain in 2 lines why do we concatenate?)</strong> This technique fuses the features at four different scales, which you can pass as a parameter to the&nbsp;<code>PSPNetClassifier</code>&nbsp; within <code>arcgis.learn</code>.</p><p> </p><p>The four levels are a defined from coarse to fine. If we pool the features at scales of (1, 3, 5, 6), first one is the most coarse feature map as it captures all the information in just 1x1 spatial location. As the spatial resolution increases, the high resolution features are also taken in account as in the case of 6x6 pyramid size.</p><p> </p><p>A <em>rule of thumb </em>is that the if the pyramid size is small i.e ~1 or 2, the model will capture the low resolution and bigger features. If the the pyramid size is bigger i.e ~6-8, the model will be able to capture high resolution features.&nbsp;If you have a large variance in the size of features to be segmented, you may want to increase the number of pyramids and the size of pyramids in the PSPNet module. On the other hand if you have similar sized features, you can stick to one or two level pyramid.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/547/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
<p>Done the improvisations and explained the colors</p><p> </p>